### ISSUE TITLE
Media added to repeated group are stored in instances folder after removing the group

### ISSUE THREAD

**Issue by kkrawczyk123** [getodk/collect#4246]
#### Software and hardware versions 
Collect v1.29, I was able to reproduce it even on 1.25 so it is an old issue Android all versions

#### Problem description
Media added to the repeated group are still visible in the instances folder after removing that group. 

#### Steps to reproduce the problem
1. Open form with repeats and media widgets (I have used ex. [AllQ4190withoutText.xml.txt](https://github.com/getodk/collect/files/5603253/AllQ4190withoutText.xml.txt) )
2. Add group
3. Add media image/video/ record audio
4. Remove group
5. Finalize/save submission
5. Go to instance folder

gif of reproduction steps:
![ezgif com-video-to-gif (41)](https://user-images.githubusercontent.com/33342488/100356366-6ac84400-2ff3-11eb-80ed-9e1f06e804bf.gif)

#### Expected behavior
As the group has been deleted, media should be also removed like after removing the response.

#### Other information 
Things you tried, stack traces, related issues, suggestions on how to fix it...

**Comment by kkrawczyk123** [comment]
I have noticed a similar issue when the following scenario:
1. Open form with any media widget (ex. All widgets)
2. Add image
3. Finalize form
4. Edit form
5. Replace image
6. Ignore changes
7. Open instances folder: both media are visible.
![ezgif com-video-to-gif (42)](https://user-images.githubusercontent.com/33342488/100368627-1c6f7100-3004-11eb-96bd-0f04e2870cd9.gif)

I was able to reproduce it on 1.28 and I wasn't on 1.27 (here it works well, only one media visible in a folder)
Possibly the fix might work for both scenarios, if not we can report a separate issue.

**Comment by grzesiek2010** [comment]
> I have noticed a similar issue when the following scenario:

Good catch! Both issues are similar but the first one as you noticed probably has existed for a long time if not always. The second one is a regression so I think it's better to have [a separate issue](https://github.com/getodk/collect/issues/4248) for it and probably add to v1.29
